thepeer-android/src/main/java/co/thepeer/sdk/ThePeer.kt

```diff
@@ -0,0 +1,183 @@
+    private var resultRegistry: ActivityResultLauncher<ThePeerParam>,
+    private var activity: AppCompatActivity,
+    private var amount: BigDecimal,
+    private var currency: String,
+    private var userReference: String,
+    private var meta: Map<String, String> = emptyMap(),
+) {
+
+    /**
+     * Config builder for initialisation parameters
+     * @param activity -> Required to launch the drop in UI activity
+     * @param amount -> The amount for the transaction
+     * @param currency -> customer preferred currency
+     * @param userReference -> Customer indexed user reference from ThePeer
+     * @param resultListener -> result event callback
+     */
+    class Builder(
+        private val activity: AppCompatActivity,
+        private var amount: BigDecimal,
+        private var currency: String,
+        private var userReference: String,
+        resultListener: ThePeerResultListener
+    ) {
+
+        var resultRegistry = activity.registerForActivityResult(
+           ThePeerResultContract(),
+           activity.activityResultRegistry
+       ) { chargeResult ->
+           when (chargeResult) {
+               is ThePeerResult.Success -> resultListener.onSuccess(chargeResult.transaction)
+               is ThePeerResult.Error -> resultListener.onError(chargeResult.error)
+               ThePeerResult.Cancelled -> resultListener.onCancelled()
+           }
+       }
+        private var publicKey = getPublicKeyFromManifest(activity)
+
+        private var meta: Map<String, String> = emptyMap()
+
+        private fun getPublicKeyFromManifest(context: Context): String {
+            val applicationInfo = context.packageManager.getApplicationInfo(
+                context.packageName,
+                PackageManager.GET_META_DATA
+            )
+            return applicationInfo.metaData?.getString(ThePeerConstants.PUBLIC_KEY_FROM_MANIFEST)
+                .orEmpty()
+        }
+
+        fun setAmount(amount: BigDecimal): Builder {
+            this.amount = amount
+            return this
+        }
+
+
+        fun setCurrency(currency: String): Builder {
+            this.currency = currency
+            return this
+        }
+
+        fun setUserReference(userReference: String): Builder {
+            this.userReference = userReference
+            return this
+        }
+
+        fun setMeta(meta: Map<String, String>): Builder {
+            this.meta = meta
+            return this
+        }
+
+        fun build(): ThePeer {
+            return ThePeer(
+                publicKey,
+                resultRegistry,
+                activity,
+                amount,
+                currency,
+                userReference,
+                meta
+            )
+
+        }
+    }
+
+
+    /**
+     * This function will be called to launch ThePeer Send Money Widget
+     */
+    fun sendMoney() {
+        val params = ThePeerParam(
+            publicKey,
+            getSdkType(ThePeerSdkType.SEND_MONEY),
```

Comment:
Send money -> Send.

We call it Send. I think I might've missed this in the files before this.😂😭